### PR TITLE
Adding internal and substrate proposals

### PR DESCRIPTION
**Pull request by AhmedKorim**
# Overview
Adding the missing proposals to the proposals kind

**Proposals to add**
- [x] AnchorCreateProposal
- [x] EvmProposal
- [x] ProposerSetUpdateProposal
- [x] RefreshVote

### CODE REVIEW

File: packages/sdk-core/src/proposals/ProposalKinds.ts [hunk 1]
```diff
@@ -1,8 +1,192 @@
...
+
+    switch (tx.type) {
+      case 'Legacy':{
+        const txValue = tx.asLegacy;
+
+        nonce = txValue.nonce.toNumber();
+        chainId = 0;
+      }
+
+        break;
```
Comment (drewstone) [inline packages/sdk-core/src/proposals/ProposalKinds.ts:46]: Aren't these breaks in the wrong place? Shouldn't they be inside each `case`?

**Review by drewstone — CHANGES_REQUESTED** [review]

File: packages/sdk-core/src/proposals/ProposalKinds.ts (hunk 1 above)
Comment (AhmedKorim) [reply packages/sdk-core/src/proposals/ProposalKinds.ts:46]: Not sure why but this happens when I run `yarn lint` will check it more

Comment (AhmedKorim) [reply packages/sdk-core/src/proposals/ProposalKinds.ts]: switched to if instead

**Comment by dutterbutter** [comment]
@AhmedKorim please address the failing tests and then should be good to merge!

File: packages/sdk-core/src/__test__/proposals.spec.ts [hunk 2]
```diff
@@ -2,15 +2,16 @@
...
 import { hexToU8a, u8aToHex } from '@polkadot/util';
 
-import { ChainType } from '../index.js';
+import { AnchorCreateProposal, ChainType, EVMProposal, ProposerSetUpdateProposal } from '../index.js';
 import { ProposalHeader } from '../proposals/ProposalHeader.js';
 import { AnchorUpdateProposal } from '../proposals/ProposalKinds.js';
 import { ResourceId } from '../proposals/ResourceId.js';
 
-describe('test various conversion functions', () => {
+describe.only('test various conversion functions', () => {
```
Comment (dutterbutter) [inline packages/sdk-core/src/__test__/proposals.spec.ts:14]: Remove .only, no?

**Pull request merged**